src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -150,7 +153,21 @@ internal ItemManager(RecentItemsManager recentItemsManager)
             var completionRules = completionService?.GetRules() ?? CompletionRules.Default;
             var completionHelper = document != null ? CompletionHelper.GetHelper(document) : _defaultCompletionHelper;
 
-            var initialListOfItemsToBeIncluded = new List<ExtendedFilterResult>();
+            // DismissIfLastCharacterDeleted should be applied only when started with Insertion, and then Deleted all characters typed.
+            // This confirms with the original VS 2010 behavior.
+            if (initialRoslynTriggerKind == CompletionTriggerKind.Insertion &&
+                data.Trigger.Reason == CompletionTriggerReason.Backspace &&
+                completionRules.DismissIfLastCharacterDeleted &&
+                session.ApplicableToSpan.GetText(data.Snapshot).Length == 0)
+            {
+                // Dismiss the session
+                return null;
+            }
+
+            var options = document?.Project.Solution.Options;
+            var highlightMatchingPortions = options?.GetOption(CompletionOptions.HighlightMatchingPortionsOfCompletionListItems, document.Project.Language) ?? true;
```

Comment:
if we don't have options, i'd default this to false.  that's a super bizarre case, and i don't think we adeqautely test that.

Reply:
Changed as requested. FYI @ivanbasov